CMakeLists.txt

```diff
@@ -40,6 +40,8 @@ ENDIF()
 
 MESSAGE( STATUS "Building Charm ${Charm_VERSION} in ${CMAKE_BUILD_TYPE} mode" )
 
+add_definitions(-DCHARM_VERSION="${Charm_VERSION}")
```

Comment:
I much much prefer generated headers (using configure_file) so that only the cpp files that need CHARM_VERSION get recompiled when the version changes. With add_definitions, every single cpp file will get recompiled, instead.